sklearn/tests/test_multiclass.py

```diff
@@ -824,6 +824,61 @@ def test_pairwise_indices():
         )
 
 
+def test_pairwise_n_features_in():
+    """Check the n_features_in_ attributes of the meta and base estimators
+
+    When the training data is a regular design matrix, everything is intuitive.
+    However, when the training data is a precomputed kernel matrix, the
+    multiclass strategy can resample the kernel matrix of the underlying base
+    estimator both row-wise and column-wise and this has a non-trivial impact
+    on the expected value for the n_features_in_ of both the meta and the base
+    estimators.
+    """
+    X, y = iris.data, iris.target
+
+    # Fitting directly on the design matrix:
+    assert X.shape == (150, 4)
+
+    clf_notprecomputed = svm.SVC(kernel="linear").fit(X, y)
+    assert clf_notprecomputed.n_features_in_ == 4
+
+    ovr_notprecomputed = OneVsRestClassifier(clf_notprecomputed).fit(X, y)
+    assert ovr_notprecomputed.n_features_in_ == 4
+    for est in ovr_notprecomputed.estimators_:
+        assert est.n_features_in_ == 4
+
+    # Fitting the same models with a precomputed kernel:
+    K = X @ X.T
+    assert K.shape == (150, 150)
+
+    ovo_notprecomputed = OneVsOneClassifier(clf_notprecomputed).fit(X, y)
+    assert ovo_notprecomputed.n_features_in_ == 4
+    for est in ovo_notprecomputed.estimators_:
+        assert est.n_features_in_ == 4
+
+    # When working with precomputed kernels we have one "feature" per training
+    # sample:
+    clf_precomputed = svm.SVC(kernel="precomputed").fit(K, y)
+    assert clf_precomputed.n_features_in_ == 150
+
+    ovr_precomputed = OneVsRestClassifier(clf_precomputed).fit(K, y)
+    assert ovr_precomputed.n_features_in_ == 150
+    for est in ovr_precomputed.estimators_:
+        assert est.n_features_in_ == 150
+
+    # This becomes really interesting with OvO and precomputed kernel together:
+    # internally, OvO will drop the samples of the classes not part of the pair
+    # of classes under consideration for a given binary classifier. Since we
+    # use a precomputed kernel, it will also drop the matching columns of the
+    # kernel matrix, and therefore we have fewer "features" as result. Since
+    # each class has 50 samples, a single OvO binary classifier works with a
+    # subkernel matrix of shape (100, 100).
```

Comment:
What happens if the number of samples is not the same in each class. I assume that we report the number of features of the first binary classifier. I am not really sure if this is the right thing to report.

Reply:
no, the meta-estimators' n_features_in_ is X.shape[1]. We don't report the n_features_in_ of the first clf any more. It's the n_features_in_ of each underlying estimator that will not be the same. (if I understood correctly :))

Reply:
Right. We missed this. It could be nice to not have the same number of samples in each class and check the consistency in the underlying binary classifier.